core/src/main/java/org/elasticsearch/transport/netty/NettyTransport.java

```diff
@@ -1156,5 +1157,14 @@ public void operationComplete(ChannelFuture future) throws Exception {
             }
             threadPool.schedule(pingSchedule, ThreadPool.Names.GENERIC, this);
         }
+
+        @Override
+        public void onFailure(Throwable t) {
+            if (lifecycle.stoppedOrClosed()) {
+                logger.trace("[{}] failed to send ping transport message", t);
+            } else {
+                logger.warn("[{}] failed to send ping transport message", t);
```

Comment:
The test failed because of a rejection exception. In that case we are pretty sure we are shutting down. In all other cases shouldn't we re-schedule the ping?